normandy/control/static/control/js/components/ActionForm.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import { reduxForm } from 'redux-form'
+import { _ } from 'underscore'
+import HeartbeatForm from './action_forms/HeartbeatForm.jsx'
+import ConsoleLogForm from './action_forms/ConsoleLogForm.jsx'
+
+class ActionForm extends React.Component {
+  constructor(props) {
+    super(props)
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    const currentProps = _.pick(this.props, 'initialValues', 'values', 'fields', 'selectedAction');
+    const incomingProps = _.pick(nextProps, 'initialValues', 'values', 'fields', 'selectedAction');
+
+    return (!_.isEqual(currentProps, incomingProps) || !_.isEqual(this.state, nextState));
```

Comment:
Performance seems fine with `shouldComponentUpdate` removed & dev tools closed. Will add back in if anyone notices otherwise.